### ISSUE TITLE
SMLLexer won't terminate on simple input

### ISSUE THREAD

**Issue by bforte** [pygments/pygments#1625]
When the `SMLLexer` gets fed the string `"exception"` it seems to loop indefinitely. Here's a simple example:

```python3
from pygments import highlight
from pygments.lexers import SMLLexer
from pygments.formatters import HtmlFormatter

code = 'exception'
print(highlight(code, SMLLexer(), HtmlFormatter()))
```

Tested on both

    $ pygmentize -V
    Pygments version 2.3.1, (c) 2006-2017 by Georg Brandl.

and

    $ pygmentize -V
    Pygments version 2.7.3, (c) 2006-2020 by Georg Brandl.

.

**Comment by birkenfeld** [comment]
Thanks for the report, this will be fixed in the next version (probably 2.7.4).

**Comment by legoktm** [comment]
For reference, this was assigned [CVE-2021-20270](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-20270).
